front/src/common/Pathfinding/TypeAndPath.tsx

```diff
@@ -65,13 +66,73 @@ export default function TypeAndPath({ zoomToFeature }: PathfindingProps) {
   const rollingStockId = useSelector(getRollingStockID);
   const [postSearch] = osrdEditoastApi.endpoints.postSearch.useMutation();
   const [postPathfinding] = osrdEditoastApi.endpoints.postPathfinding.useMutation();
-  const { t } = useTranslation('operationalStudies/manageTrainSchedule');
+  const { t: tManageTrainSchedule } = useTranslation('operationalStudies/manageTrainSchedule');
+  const { t: tTypeAndPath } = useTranslation('common/typeAndPath');
+
   const osrdActions = useOsrdConfActions();
 
+  const [searchResults, setSearchResults] = useState<SearchResultItemOperationalPoint[]>([]);
+  const [searchState, setSearch] = useState('');
+  const mainOperationalPointsCHCodes = ['', '00', 'BV'];
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const debouncedSearchTerm = useDebounce(searchState, 300);
   const debouncedInputText = useDebounce(inputText.trimEnd(), 500);
 
-  const handleInput = (text: string) => {
-    setInputText(text.trimStart());
+  const activeElement = document.activeElement as HTMLInputElement;
+  const cursorIndex = activeElement.selectionStart || 0;
+  const sortedSearchResults = [...searchResults].sort((a, b) => a.name.localeCompare(b.name));
+  const [initialCursorPositionRem, setInitialCursorPositionRem] = useState(0);
+  const [trigramCount, setTrigramCount] = useState(0);
+
+  const handleInput = (text: string, cursorPosition: number) => {
+    const trimmedTextStart = text.trimStart();
+    setInputText(trimmedTextStart);
+
+    const isCursorSurroundedBySpaces =
+      text[cursorPosition - 1] === ' ' &&
+      (text[cursorPosition] === ' ' || cursorPosition === text.length);
```

Comment:
maybe put those transformation function in an utils file

Reply:
done